### ISSUE TITLE
Build failure: ollama when specifying acceleration as rocm

### ISSUE THREAD

**Issue by codebam** [NixOS/nixpkgs#306654]
### Steps To Reproduce

1.
```nix
  services.ollama = {
    enable = true;
    acceleration = "rocm";
  };
```

### Build log

```
log here if short otherwise a link to a gist
```

https://gist.github.com/codebam/be066269c54cd2522b1f7c9b2c3f0d7d

### Additional context

Using Radeon 7900XTX

### Notify maintainers

@abysssol @dit7ya @elohmeier

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by codebam** [comment]
After looking at this https://discourse.cmake.org/t/possible-regression-related-to-interface-include-directories-in-2-29-1/10705

I think this may be resolved in the next version of cmake?

**Comment by codebam** [comment]
https://github.com/NixOS/nixpkgs/pull/306684

**Comment by abysssol** [comment]
The main issue is #305641, fixed by #306375. I ran ollama's tests on `master`, and they all passed. It'll just take some time to make it into unstable.
